### PR TITLE
Refactor the big gitlab CI yaml file

### PR DESCRIPTION
**Pull request by sandreim**
This PR splits it in separate per stage files

### CODE REVIEW

**Review by drahnr — DISMISSED** [review]

**Comment by sergejparity** [comment]
Hi @sandreim It would be great if you can move those `./ci/*` files into `./scripts/ci/` directory. Actually we recently started unification of the CI related files in `paritytech` org repositories. More about it: https://github.com/paritytech/ci_cd/issues/313

**Comment by TriplEight** [comment]
Funnily enough, we've exactly the same planned. Thanks for helping out here!

**Comment by TriplEight** [comment]
Looking what might be wrong with CI config with https://gitlab.parity.io/parity/polkadot/-/ci/editor and it shows nothing useful sadly.

**Comment by TriplEight** [comment]
As per [docs](https://docs.gitlab.com/ee/ci/yaml/#include)

> You can [nest](https://docs.gitlab.com/ee/ci/yaml/includes.html#use-nested-includes) up to 100 includes, but you can’t have duplicate includes. In [GitLab 12.4 and later](https://gitlab.com/gitlab-org/gitlab/-/issues/28212), the time limit to resolve all files is 30 seconds.

**Comment by sergejparity** [comment]
Also there could be a problem with [**In CI, YAML anchors don't work with include**](https://gitlab.com/gitlab-org/gitlab/-/issues/5467).

File: ci/build.yml [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  stage:                           build
+  <<:                              *build-push-image
+  <<:                              *zombienet-refs
+  variables:
+    <<:                            *image-variables
+    # scripts/dockerfiles/collator_injected.Dockerfile
+    DOCKERFILE:                    dockerfiles/collator_injected.Dockerfile
+    IMAGE_NAME:                    docker.io/paritypr/colander
+  needs:
+    - job:                         build-adder-collator
```
Comment (TriplEight) [inline ci/build.yml:150]: `build-adder-collator` is `need`ed on L150

File: ci/build.yml [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  variables:
+    RUNTIME:                       kusama
+
+update_westend_weights:
+  <<:                              *update-weights
+  variables:
+    RUNTIME:                       westend
+
+
+build-adder-collator:
```
Comment (TriplEight) [inline ci/build.yml:211]: ...and defined on L211

**Review by TriplEight — CHANGES_REQUESTED** [review]

File: ci/build.yml (hunk 2 above)
Comment (TriplEight) [reply ci/build.yml:211]: moreover it should be on the previous stage

**Comment by TriplEight** [comment]
> Also there could be a problem with [**In CI, YAML anchors don't work with include**](https://gitlab.com/gitlab-org/gitlab/-/issues/5467).

this.
Here's how to workaround it: all YAML anchors should be rewritten to https://docs.gitlab.com/ee/ci/yaml/yaml_optimization.html#reference-tags

**Comment by TriplEight** [comment]
Oh that's a bummer, we can't use gitlab reference tags with `rules` keyword as it was included only in 14.3
image
We're blocked here with https://github.com/paritytech/ci_cd/issues/171
cc @paritytech/ci

**Comment by TriplEight** [comment]
I've made a testing project to lint this PR easier. https://gitlab.parity.io/TriplEight/polkadot-ci/-/merge_requests

**Review by chevdor — APPROVED** [review]

**Comment by TriplEight** [comment]
We're unblocked, GitLab is 14.9 now.

**Comment by sandreim** [comment]
closing in favor of https://github.com/paritytech/polkadot/pull/6031

**Pull request closed without merging**